Sources/Datadog/RUM/RUMMonitor.swift

```diff
@@ -164,39 +164,39 @@ public class RUMMonitor: DDRUMMonitor, RUMCommandSubscriber {
     // MARK: - Initialization
 
     /// Initializes the Datadog RUM Monitor.
-    public static func initialize(in core: DatadogCoreProtocol = defaultDatadogCore) -> DDRUMMonitor {
+    // swiftlint:disable:next function_default_parameter_at_end
+    public static func initialize(in core: DatadogCoreProtocol = defaultDatadogCore, configuration: RUMConfiguration) throws {
         do {
             if core is NOPDatadogCore {
                 throw ProgrammerError(
                     description: "`Datadog.initialize()` must be called prior to `RUMMonitor.initialize()`."
                 )
             }
-            if Global.rum is RUMMonitor {
+
+            let feature = DatadogRUMFeature(in: core, configuration: configuration)
+            core.v1.feature(URLSessionAutoInstrumentation.self)?.publish(to: feature.monitor)
+            try core.register(feature: feature)
```

Comment:
q/ I'm quite surprised to see `RUMMonitor` registering the RUM feature to core 🤔💭. I was thinking we want to decouple both (feature registration + data collector retrieval):
```swift
RUM.initialize(with: configuration, in: core)

let monitor = RUMMonitor.shared(in: core)
```
☝️ Similar to how we do it for Session Replay. I don't remember if we discussed it before 🤔?

Reply:
That's a good point! It would make sense to split the 2 indeed. I've updated [RUMM-2922](https://datadoghq.atlassian.net/browse/RUMM-2922) to include that public API change 👍